src/pykiso/test_coordinator/test_case.py

```diff
@@ -157,6 +157,21 @@ def __init__(
                 "BasicTest does not support test timeouts, it will be discarded"
             )
 
+        # Store additional data fetched during test for
+        # the step-report (even if not activated)
+        self.step_report_header = {}
+        # Message for step in report
+        self.step_report_message = ""
+        # Test succeed flag
+        # Set to false if one or more steps fail
+        self.step_report_succeed = True
+        # Error message on step fail
+        self.step_report_last_error_message = ""
+        # Flag used for stopping interruptions on first fail for reporting
+        self.step_report_continue_on_error = False
+        # Current report table
+        self.step_report_current_table = None
```

Comment:
This adds a lot of attributes for a single functionality. You could wrap them all in an intermediary object, e.g.:
```
@dataclass
class StepReportConfig:
    header: dict = field(default_factory=dict)
    message: str = ""
    success: bool = True
    last_error_message: str = ""
    continue_on_error: bool = False # or use the failfast value?
    current_table: Optional[WhatType?] = None
```

And here you would just do:
```self.step_report = StepReportConfig()```